sampler.go

```diff
@@ -112,21 +113,34 @@ func registerSampler(m map[string]setupFunc, app *kingpin.Application, name stri
 		}
 		c := storepb.NewWritableProfileStoreClient(conn)
 
-		var samplerOpts []SamplerOption
+		db := store.NewGRPCAppendable(logger, c)
+		scrapeManager := scrape.NewManager(log.With(log.NewNopLogger(), "component", "scrape-manager"), db)
+
+		samplerOpts := []SamplerOption{
+			SamplerScraper(scrapeManager),
+		}
 		if len(*targets) > 0 {
 			samplerOpts = append(samplerOpts, SamplerTargets(*targets))
 		} else {
 			samplerOpts = append(samplerOpts, SamplerConfig(*configFile))
 		}
 
-		s, err := NewSampler(store.NewGRPCAppendable(logger, c), reloaders, samplerOpts...)
+		s, err := NewSampler(db, reloaders, samplerOpts...)
 		if err != nil {
 			return nil, err
 		}
 		if err := s.Run(context.TODO(), g, reloadCh); err != nil {
 			return nil, err
 		}
 
+		api := conprofapi.New(logger, reg, NopQuerier{}, reloadCh, 64*1024*1024,
+			func(ctx context.Context) conprofapi.TargetRetriever {
+				return scrapeManager
+			},
+		)
+		const apiPrefix = "/api/v1/"
```

Comment:
Nitpick: Is this constant something you can define in the conprofapi package?

Reply:
Yes. I'll take care of that along with some other things to refactor in there.

Reply:
Nice! :niceee: